### PR TITLE
doc: add zigbee helper library page

### PR DESCRIPTION
**Pull request by greg-fer**
Add zigbee helper functions and rework structure.
(KRKNWK-6713)

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@wbober , this PR needs the following updates:

- Review of the new pages, including the helper functions one.
- Update of the ug_zigbee_configuring_components.rst with information about how to configure these helper functions.

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+Zigbee application helper functions
+###################################
+
+|NCS| provides several helper functions that are intended to facilitate the end user application implementation when using the :ref:`nrfxlib:zboss` library.
+They implement the default behavior of a Zigbee device and are used in :ref:`zigbee_samples` in this SDK.
+
+.. _zarco_signal_handler:
+
+Zigbee default signal handler
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:11]: I think this should be elevated to a section on it's own.

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 2]
```diff
@@ -0,0 +1,322 @@
+﻿.. _lib_zigbee_app_helper_functions:
+
+Zigbee application helper functions
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:3]: I think we should remove "Zigbee application helper functions" and just discuss the signal handler here.

File: doc/nrf/libraries/lib_zigbee.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+|zigbee_description|
+
+See the :ref:`ug_zigbee` user guide for an overview of the technology.
+For information on how to use the supplied libraries, see :ref:`ug_zigbee_configuring_components`.
+
+.. toctree::
+   :maxdepth: 1
+   :glob:
+
+   zigbee/lib_zigbee_app_helper_functions
```
Comment (wbober) [inline doc/nrf/libraries/lib_zigbee.rst:15]: As below. I'd discuss the default signal handler.

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+|NCS| provides several helper functions that are intended to facilitate the end user application implementation when using the :ref:`nrfxlib:zboss` library.
+They implement the default behavior of a Zigbee device and are used in :ref:`zigbee_samples` in this SDK.
+
+.. _zarco_signal_handler:
+
+Zigbee default signal handler
+*****************************
+
+The Zigbee default signal handler is a function called from the :cpp:func:`zboss_signal_handler()` function that is mandatory for each application.
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:14]: I'd rephrase a bit.

```
ZBOSS stack interacts with the user application by invoking `zboss_signal_handler()` function 
whenever a stack event, such as network steering, occurs. It is mandator to define 
`zboss_signal_handler()` in the application. Because [...] signals. This function can be 
called in application's `zboss_signal_handler()` simplifying implementation.
```

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+        if (bufid)
+        {
+            zb_buf_free(bufid);
+        }
+    }
+
+
+.. _zarco_signal_handler_startup:
+
+Zigbee stack startup without autostart
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:98]: Rename to "Behavior on stack start" or something like that :)

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+======================================
+
+Whenever :cpp:func:`zigbee_default_signal_handler()` is used, the stack should be started using a call to :cpp:func:`zboss_start_no_autostart()`:
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:101]: Remove text between lines 101 to 119 (exclusive).

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+
+* It allows you to ensure better granularity in customizing the default procedure.
+* You gain a better understanding of procedures executed by the stack, because all procedures specified by the Zigbee Base Device Behavior specification are explicitly started inside the default signal handler and generate a separate signal once finished.
+
+.. figure:: /images/zigbee_signal_handler_00_stack_init.png
+   :alt: Zigbee stack startup without autostart
+
+   Zigbee stack startup without autostart
+
+When the stack is started through :cpp:func:`zboss_start_no_autostart()`, the stack generates the following signals:
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:119]: Rephrase "When the stack is started through `zigbee_enable()`, the stack generates the following signals:"

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+  * The procedure to join or rejoin the network can be restarted by calling :cpp:func:`user_input_indicate()`, but it needs to be implemented in the application (for example, by calling :cpp:func:`user_input_indicate()` when a button is pressed).
+    The procedure will be restarted only if the device does not join and the procedure is not running.
+
+The Zigbee rejoin procedure retries to join a network with each try after a specified amount of time: ``2^n`` seconds, where ``n`` is the number of retries.
+
+The period is limited to 15 minutes if the result is higher than that.
+
+* When :cpp:func:`start_network_rejoin()` is called, the rejoin procedure is started, and depending on the device role:
+
+  * For the end device, the application alarm is scheduled with ``stop_network_rejoin(ZB_TRUE)``, to be called after the amount of time specified in ``ZB_DEV_REJOIN_TIMEOUT_MS``.
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:267]: @sebastiandraus does ```ZB_DEV_REJOIN_TIMEOUT_MS``` is from KConfig options. If, so we should use it here.

File: doc/nrf/libraries/lib_zigbee.rst [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿.. _lib_zigbee:
+
+Libraries for Zigbee
+####################
+
+|zigbee_description|
+
+See the :ref:`ug_zigbee` user guide for an overview of the technology.
+For information on how to use the supplied libraries, see :ref:`ug_zigbee_configuring_components`.
```
Comment (wbober) [inline doc/nrf/libraries/lib_zigbee.rst:9]:
```
Zigbee libraries are a set of components that can be directly reused in Zigbee applications.
```

File: doc/nrf/ug_zigbee_configuring_components.rst [hunk 10]
```diff
@@ -0,0 +1,31 @@
+﻿.. _ug_zigbee_configuring_components:
+
+Configuring Zigbee components in |NCS|
```
Comment (wbober) [inline doc/nrf/ug_zigbee_configuring_components.rst:3]: I think we need to decide on a single name for this. So far I see several
1) Zigbee libraries
2) Zigbee helper functions
3) Zigbee components

I think the best name is either 1) or "Zigbee Application Components".

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst [hunk 11]
```diff
@@ -0,0 +1,322 @@
...
+
+.. warning::
+    The Zigbee network rejoin procedure is managed from multiple signals in :cpp:func:`default_signal_handler()`.
+    If the application controls the network joining, rejoining, or leaving, each signal in which the Zigbee network rejoin procedure is managed should be handled in the application.
+    In this case, :cpp:func:`user_input_indicate()` must not be called.
+
+.. _zarco_sleep:
+
+Zigbee stack sleep routines
+***************************
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:286]: I think most of the text in this section can be removed as it is already discussed here https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_zigbee_configuring.html#sleepy-end-device-behavior

Please consult with @tomchy.

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst (hunk 8 above)
Comment (greg-fer) [reply doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst:267]: @sebastiandraus , any updates about this?

File: doc/nrf/ug_zigbee_configuring_components.rst [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+This page lists options and steps required for configuring the available components:
+
+.. contents::
+    :local:
+    :depth: 1
+
+.. _ug_zigbee_configuring_components_ota:
+
+Configuring application helper functions
```
Comment (greg-fer) [inline doc/nrf/ug_zigbee_configuring_components.rst:16]: @wbober , any feedback about this section? It needs new NCS input.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 13]
```diff
@@ -0,0 +1,315 @@
...
+
+.. warning::
+    The Zigbee network rejoin procedure is managed from multiple signals in :cpp:func:`default_signal_handler()`.
+    If the application controls the network joining, rejoining, or leaving, each signal in which the Zigbee network rejoin procedure is managed should be handled in the application.
+    In this case, :cpp:func:`user_input_indicate()` must not be called.
+
+????????????????????????????????????????????????????
+.. _zarco_sleep:
+
+Zigbee stack sleep routines
```
Comment (wbober) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:278]: @tomchy or @sebastiandraus please review this.

File: doc/nrf/ug_zigbee_configuring_components.rst (hunk 12 above)
Comment (wbober) [reply doc/nrf/ug_zigbee_configuring_components.rst:16]: @tomchy or @sebastiandraus please provide your input on KConfig options for default signal handler.

File: doc/nrf/images/zigbee_signal_handler_00_stack_init.uml [hunk 14]
```diff
@@ -0,0 +1,7 @@
+@startuml
+
+start
+
+:zboss_start_no_autostart();
```
Comment (tomchy) [inline doc/nrf/images/zigbee_signal_handler_00_stack_init.uml:5]: consider: mentioning `zigbee_enable()` that calls `zboss_start_no_autorstart()`

File: doc/nrf/images/zigbee_signal_handler_01_production_config.uml [hunk 15]
```diff
@@ -0,0 +1,9 @@
+@startuml
+
+:ZB_ZDO_SIGNAL_PRODUCTION_CONFIG_READY;
+
+:NRF_LOG_INFO(load_status);
```
Comment (tomchy) [inline doc/nrf/images/zigbee_signal_handler_01_production_config.uml:5]: `NRF_LOG_INFO` unavailable in NCS.
According to our newest guidelines - should it be replaced with a simple comment?

File: doc/nrf/images/zigbee_signal_handler_07_idle.uml [hunk 16]
```diff
@@ -0,0 +1,18 @@
+@startuml
+
+:zboss_main_loop_iteration();
+
+if (scheduler_queue_empty?) then (yes)
+if ((inactivity_time > sleep_threshold) and (rx_off_when_idle)) then (yes)
```
Comment (tomchy) [inline doc/nrf/images/zigbee_signal_handler_07_idle.uml:6]: To be improved:
 - `zb_osif_go_idle` left unimplemented in ZBOSS
 - `ZB_COMMON_SIGNAL_CAN_SLEEP` is even if `rx_off_when_idle` if set to false.

File: doc/nrf/images/zigbee_signal_handler_08_deep_sleep.uml [hunk 17]
```diff
@@ -0,0 +1,14 @@
+@startuml
+
+:ZB_COMMON_SIGNAL_CAN_SLEEP;
+partition zb_sleep_now() {
+partition zb_nrf52840_sleep() #AntiqueWhite {
+        :zb_nrf52840_priph_disable();
```
Comment (tomchy) [inline doc/nrf/images/zigbee_signal_handler_08_deep_sleep.uml:6]: To be improved:
 - All functions starting from `zb_nrf52840_` prefix were renamed in NCS.
 - Both `periph_enable` and `periph_disable` removed from NCS: ZBOSS no longer manages power state of shared resources. It disables `RADIO` and `TIMER/COUNTER`, but not from the `sleep` function, but from the core code.
 - `zb_osif_wait_for_event()` replaced by `zigbee_event_poll(tmo)`

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 18]
```diff
@@ -0,0 +1,303 @@
...
+        zb_ret_t                   status = ZB_GET_APP_SIGNAL_STATUS(bufid);
+
+        switch (sig)
+        {
+            case ZB_BDB_SIGNAL_DEVICE_REBOOT:
+                /* fall-through */
+            case ZB_BDB_SIGNAL_STEERING:
+                if (status == RET_OK)
+                {
+                    NRF_LOG_INFO("Joined network successfully");
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:69]: `NRF_LOG_INFO` unavailable in NCS

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 19]
```diff
@@ -0,0 +1,303 @@
...
+    In this case, :cpp:func:`user_input_indicate()` must not be called.
+
+.. _zarco_sleep:
+
+Zigbee stack sleep routines
+***************************
+
+The Zigbee stack provides the following mechanisms to reduce power consumption:
+
+* On Zigbee end devices, routers and coordinators:
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:271]: To be removed (no IDLE state in NCS).

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 20]
```diff
@@ -0,0 +1,303 @@
...
+
+  * The stack calls the Power Management API (through :cpp:func:`zb_osif_wait_for_event`) if the Zigbee scheduler queue is empty.
+
+* On Zigbee sleepy end devices:
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:275]: To be simplified. Applies to all devices, starting from `If there are no delayed callbacks...`, with one exception: routers and coordinators do not turn off their timer.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 21]
```diff
@@ -0,0 +1,303 @@
...
+
+On Zigbee sleepy end devices, the ``sleep_threshold`` period lasts approximately 15 ms by default and can be modified by using the ``zb_sleep_set_threshold`` API.
+
+If the device implements a Zigbee sleepy end device, the ``ZB_COMMON_SIGNAL_CAN_SLEEP`` signal is generated whenever the device can be put into the deep sleep mode.
+This signal can also be handled using the Zigbee default signal handler.
+If so, it will allow the Zigbee stack to enter the deep sleep state.
+
+If the default behavior is not applicable for the application, you can customize the sleep functionality by using the following options:
+
+* Overwriting the @ref zb_osif_go_idle weak function and implementing a custom logic for handling the stack idle state.
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:288]: To be removed: IDLE state not present.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 22]
```diff
@@ -0,0 +1,303 @@
...
+
+  .. figure:: /images/zigbee_signal_handler_07_idle.png
+     :alt: Implementing a custom logic for handling the stack idle state
+
+     Implementing a custom logic for handling the stack idle state
+
+* Overwriting the :cpp:func:`zb_nrf52840_sleep` weak function and implementing a custom logic for putting the stack into the deep sleep mode.
+
+  * As part of this option, overwriting :cpp:func:`zb_nrf52840_periph_disable` and :cpp:func:`zb_nrf52840_periph_enable` weak functions to change the set of peripheral devices, whose power is managed by the Zigbee stack.
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:297]: To be removed: `periph` functions left unimplemented/not called.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 23]
```diff
@@ -0,0 +1,303 @@
...
+  * The stack calls the Power Management API (through :cpp:func:`zb_osif_wait_for_event`) if the Zigbee scheduler queue is empty.
+
+* On Zigbee sleepy end devices:
+
+  * If there are no immediate callbacks scheduled through the Zigbee scheduler, the stack calls the Power Management API (through :cpp:func:`zb_osif_wait_for_event`).
+  * If there are no delayed callbacks (alarms) scheduled for the next ``sleep_threshold`` period, the stack switches off all peripheral devices related to the Zigbee stack before calling the Power Management API.
+
+On Zigbee sleepy end devices, the ``sleep_threshold`` period lasts approximately 15 ms by default and can be modified by using the ``zb_sleep_set_threshold`` API.
+
+If the device implements a Zigbee sleepy end device, the ``ZB_COMMON_SIGNAL_CAN_SLEEP`` signal is generated whenever the device can be put into the deep sleep mode.
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:282]: To be improved:
 - The reasoning behind the sleep signal is no longer focused on energy consumption. We implement it for all device types, so the ZBOSS task does use CPU time.
 - The sleep threshold applies to all device types.
 - ZBOSS OSIF does not call Power Management API.
 - ZBOSS OSIF does not manage shared resources. The Radio and timer state are managed by ZBOSS core.

File: doc/nrf/ug_zigbee_configuring_components.rst [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+
+.. _ug_zigbee_configuring_components_ota:
+
+Configuring default signal handler
+**********************************
+
+The default signal handler provides a default logic to handle ZBOSS stack signals.
+For more information, see :ref:`lib_zigbee_signal_handler`.
+
+The default signal handler can be enabled and configured using the following Kconfig options:
```
Comment (tomchy) [inline doc/nrf/ug_zigbee_configuring_components.rst:22]: No Kconfig options available. It is used by calling the `zigbee_default_signal_handler` inside aplication's `zboss_zignal_handler` implementation.

File: doc/nrf/ug_zigbee_configuring_components.rst [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+
+* (mandatory option 1) - This option must be enabled to use the DFU OTA.
+* (optional option 1) - This option allows to do XYZ.
+* (optional option 2) - This option allows to do XYZ.
+
+When configuring default signal handler, take into account the following principles:
```
Comment (tomchy) [inline doc/nrf/ug_zigbee_configuring_components.rst:28]: We may rewrite it if you want to, to mention the features that are left unimplemented in default signal handler (such as parsing the production configuration, finding and binding, commissioning other than a network steering with default TCLK).

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst (hunk 18 above)
Comment (greg-fer) [reply doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:69]: Replaced by a comment field.

File: doc/nrf/ug_zigbee_configuring_components.rst (hunk 24 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee_configuring_components.rst]: Reworked this part.

File: doc/nrf/ug_zigbee_configuring_components.rst (hunk 25 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee_configuring_components.rst]: Not sure, to be honest...

File: doc/nrf/ug_zigbee_configuring_components.rst (hunk 10 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee_configuring_components.rst]: Let's stick to 1) for now.

File: doc/nrf/ug_zigbee_configuring_components.rst (hunk 12 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee_configuring_components.rst]: Talked to Tomasz.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst (hunk 13 above)
Comment (greg-fer) [reply doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:269]: Reviewed by Tomasz.

**Comment by greg-fer** [comment]
@tomchy , feel free to implement the changes to .uml files you mentioned in comments and Teams, as we agreed on Teams.

File: doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst (hunk 8 above)
Comment (sebastiandraus) [reply doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst]: No, the `ZB_DEV_REJOIN_TIMEOUT_MS` does not come from the KConfig.

File: doc/nrf/images/zigbee_signal_handler_05_formation.uml [hunk 26]
```diff
@@ -0,0 +1,13 @@
+@startuml
+
+:ZB_BDB_SIGNAL_FORMATION;
+
+if (status == RET_OK) then (no)
+    :bdb_start_top_level_commissioning(ZB_BDB_NETWORK_FORMATION);
+else (yes)
+    :start_network_rejoin();
```
Comment (sebastiandraus) [inline doc/nrf/images/zigbee_signal_handler_05_formation.uml:8]: It's _bdb_start_top_level_commissioning(ZB_BDB_NETWORK_STEERING)_

File: doc/nrf/images/zigbee_signal_handler_09_leave.uml [hunk 27]
```diff
@@ -0,0 +1,11 @@
+@startuml
+
+:ZB_ZDO_SIGNAL_LEAVE;
+
```
Comment (sebastiandraus) [inline doc/nrf/images/zigbee_signal_handler_09_leave.uml:4]: It has been changed, now the coordinator forms network instead of calling the `start_network_rejoin()`.

This should look more like this:

> if (status == RET_OK) then (no)
else (yes)
        if (role == coordinator) then (no)
                :start_network_rejoin();
        else (yes)
                bdb_start_top_level_commissioning(ZB_BDB_NETWORK_FORMATION);
         endif
endif

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 28]
```diff
@@ -0,0 +1,296 @@
...
+
+    void zboss_signal_handler(zb_bufid_t bufid)
+    {
+        zb_zdo_app_signal_hdr_t  * p_sg_p = NULL;
+        zb_zdo_app_signal_type_t   sig    = zb_get_app_signal(bufid, &p_sg_p);
+        zb_ret_t                   status = ZB_GET_APP_SIGNAL_STATUS(bufid);
+
+        switch (sig)
+        {
+            case ZB_BDB_SIGNAL_DEVICE_REBOOT:
```
Comment (sebastiandraus) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:67]: I would add here also `ZB_BDB_SIGNAL_DEVICE_FIRST_START` because `start_network_rejoin()` is used for this signal in the `default_signal_handler()`.
Similarly, `ZB_ZDO_SIGNAL_LEAVE` should be added here. If default behavior is to be replaced, it should be replaced completely.

File: doc/nrf/images/zigbee_signal_handler_05_formation.uml (hunk 26 above)
Comment (tomchy) [reply doc/nrf/images/zigbee_signal_handler_05_formation.uml:8]: Are you sure about that?
```
		LOG_INF("Device started for the first time");
		if (status == RET_OK) {
			if (role != ZB_NWK_DEVICE_TYPE_COORDINATOR) {
				LOG_INF("Start network steering");
				start_network_rejoin();
			} else {
				LOG_INF("Start network formation");
				comm_status = bdb_start_top_level_commissioning(
					ZB_BDB_NETWORK_FORMATION);
			}
		} else {
			LOG_ERR("Failed to initialize Zigbee stack (status: %d)",
				status);
		}
		break;
```

Comment (tomchy) [reply doc/nrf/images/zigbee_signal_handler_05_formation.uml:8]: Agree. The current flow should have the network steering there:
```
if (status == RET_OK) {
	comm_status = bdb_start_top_level_commissioning(ZB_BDB_NETWORK_STEERING);
} else {
	ret_code = ZB_SCHEDULE_APP_ALARM(bdb_start_top_level_commissioning, ZB_BDB_NETWORK_FORMATION, ZB_TIME_ONE_SECOND);
}
```

File: doc/nrf/images/zigbee_signal_handler_09_leave.uml (hunk 27 above)
Comment (tomchy) [reply doc/nrf/images/zigbee_signal_handler_09_leave.uml:4]: I cannot find that behavior inside the source code.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst (hunk 28 above)
Comment (tomchy) [reply doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:67]: What do you think about replacing `bdb_start_top_level_commissioning` with `start/stop_network_rejoin()`?

File: doc/nrf/images/zigbee_signal_handler_09_leave.uml (hunk 27 above)
Comment (sebastiandraus) [reply doc/nrf/images/zigbee_signal_handler_09_leave.uml:4]: The change was merged just yesterday, please check the most current master branch.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst (hunk 28 above)
Comment (sebastiandraus) [reply doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:67]: It depends on what is the purpose of this section. Do we want to have implementation which does not use `start/stop_network_rejoin()` or use the same implementation as `default_signal_handler()`?

File: doc/nrf/images/zigbee_signal_handler_09_leave.uml (hunk 27 above)
Comment (tomchy) [reply doc/nrf/images/zigbee_signal_handler_09_leave.uml:4]: And what was the reasoning behind this change? Can Zigbee coordinator leave the network it has formed?

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst (hunk 28 above)
Comment (tomchy) [reply doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:67]: It is already blended in default signal handler, so I do not see much harm in calling it here. Additionally - there is no Kconfig switch to disable the rejoin process, so it has to be managed by the app if it does not override all signal handlers.

File: doc/nrf/images/zigbee_signal_handler_09_leave.uml (hunk 27 above)
Comment (sebastiandraus) [reply doc/nrf/images/zigbee_signal_handler_09_leave.uml:4]: Yes it can, `zb_bdb_reset_via_local_action()` in the end indicates `leave` signal.

File: doc/nrf/images/zigbee_signal_handler_05_formation.uml (hunk 26 above)
Comment (tomchy) [reply doc/nrf/images/zigbee_signal_handler_05_formation.uml:8]: Still not fixed.

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst (hunk 28 above)
Comment (sebastiandraus) [reply doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst]: This section is now splitted to two subsections and only `bdb_start_top_level_commissioning()` used in doc snippets.

File: doc/nrf/images/zigbee_signal_handler_05_formation.uml (hunk 26 above)
Comment (sebastiandraus) [reply doc/nrf/images/zigbee_signal_handler_05_formation.uml]: Fixed

File: doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst [hunk 29]
```diff
@@ -0,0 +1,403 @@
...
+                        break;
+
+                    default:
+                        /* Unrecognised status of F&B procedure. */
+                        break;
+                    }
+                } else {
+                    /* F&B procedure failed. */
+                }
+                break;
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst:103]: Incomplete code snippet:
 - The switch-case structure does not have the closing bracket.
 - There is no call to the default signal handler.
 - The buffer is not freed.

The alternative to adding `default_signal_handler` call is to write in the text above, that this snippet should be placed above the minimal implementation.
